lib/uktt/parser/json_api.rb

```diff
@@ -0,0 +1,77 @@
+module Uktt
+  class Parser
+    class JsonApi
+      def initialize(body)
+        @body = JSON.parse(body)
+      end
+
+      def parse
+        if data.is_a?(Hash)
+          parse_resource(data)
+        elsif data.is_a?(Array)
+          parse_collection(data)
+        else
+          data
+        end
+      end
+
+      def errors
+        @body['error'] || @body['errors']&.map { |error| error['detail'] }&.join(', ')
+      end
+
+      private
+
+      def data
+        @data = @body['data']
+      end
+
+      def parse_resource(resource)
+        result = {}
+
+        parse_top_level_attributes!(resource, result)
```

Comment:
Looking at the `parse_top_level_attributes !` method, and the fact that you initialize result with an empty hash, why not do 
```ruby
result = attributes['attributes'] ?
```